File: algorithmic_toolbox/week2_algorithmic_warmup/4_least_common_multiple/LCM.java
import java.util.Scanner;

public class LCM {
  private static long lcm_naive(int a, int b) {
    for (long l = 1; l <= (long) a * b; ++l)
      if (l % a == 0 && l % b == 0)
        return l;

    return (long) a * b;
  }

  private static int gcd(int a, int b) {
    int min = Math.min(a, b);
    int max = Math.max(a, b);

    return min == 0 ? max : gcd(max % min, min);
  }

  private static long lcm(int a, int b) {
    return (long) a * b / gcd(a, b);
  }

  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    int a = scanner.nextInt();
    int b = scanner.nextInt();
    scanner.close();

    System.out.println(lcm(a, b));
    System.out.println(lcm_naive(a, b));
  }
}

File: algorithmic_toolbox/week2_algorithmic_warmup/4_least_common_multiple/lcm.py
import sys


def lcm_naive(a: int, b: int) -> int:
    for lcm in range(1, a * b + 1):
        if lcm % a == lcm % b == 0:
            return lcm

    return a * b


def gcd(a: int, b: int) -> int:
    _min = min(a, b)
    _max = max(a, b)

    return _max if _min == 0 else gcd(_max % _min, _min)


def lcm(a: int, b: int) -> int:
    return a * b // gcd(a, b)


if __name__ == "__main__":
    a, b = map(int, sys.stdin.read().split())
    print(lcm(a, b))
    print(lcm_naive(a, b))
